From c18c057cae623530dbf813af42ad38da8cd6863b Mon Sep 17 00:00:00 2001
From: Hai Zhang <zhanghai@google.com>
Date: Fri, 12 Nov 2021 13:39:04 +0000
Subject: [PATCH] Allow bypassing qualification for system speech recognizer
 when testing.

Bug: 202312327
Test: atest RoleManagerTest#systemRoleDoesNotOverrideUserRevokedPermission
Change-Id: Iab245874168d014083b904081adb65831c9e9c6c
Merged-In: Iab245874168d014083b904081adb65831c9e9c6c
(cherry picked from commit 6352f97b38b45c40113014cc68d85360616a18ce)
Merged-In:Iab245874168d014083b904081adb65831c9e9c6c
---
 PermissionController/res/xml/roles.xml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/PermissionController/res/xml/roles.xml b/PermissionController/res/xml/roles.xml
index 30052ad49..1d24208bc 100644
--- a/PermissionController/res/xml/roles.xml
+++ b/PermissionController/res/xml/roles.xml
@@ -568,6 +568,7 @@
 
     <role
         name="android.app.role.SYSTEM_SPEECH_RECOGNIZER"
+        allowBypassingQualification="true"
         defaultHolders="config_systemSpeechRecognizer"
         exclusive="true"
         minSdkVersion="31"
-- 
2.33.1

